packages/data/src/registry.js

```diff
@@ -298,6 +301,55 @@ export function createRegistry( storeConfigs = {} ) {
 		return get( namespaces, [ reducerKey, 'actions' ] );
 	}
 
+	/**
+	 * Setup persistence for the current registry.
+	 *
+	 * @param {string} storageKey The storage key.
+	 */
+	function setupPersistence( storageKey ) {
+		const persistenceStorage = getPersistenceStorage();
+
+		// Load initially persisted value
+		let previousValue = null;
+		const persistedString = persistenceStorage.getItem( storageKey );
+		if ( persistedString ) {
+			const persistedData = JSON.parse( persistedString );
+			Object.entries( namespaces ).forEach( ( [ reducerKey, { reducer, store, persist } ] ) => {
+				if ( ! persist ) {
+					return;
+				}
+
+				const persistedState = {
+					...reducer( undefined, { type: '@@gutenberg/init' } ),
+					...get( persistedData, reducerKey ),
+				};
+
+				store.dispatch( {
+					type: 'REDUX_REHYDRATE',
+					payload: persistedState,
+				} );
+			} );
+
+			// Avoid initial save.
+			previousValue = persistedData;
+		}
+
+		// Persist updated preferences
+		subscribe( () => {
+			const newValue = Object.entries( namespaces )
+				.filter( ( [ , { persist } ] ) => persist )
+				.reduce( ( memo, [ reducerKey, { reducer, store } ] ) => {
+					memo[ reducerKey ] = reducer( store.getState(), { type: 'SERIALIZE' } );
+					return memo;
+				}, {} );
+
+			if ( newValue !== previousValue ) {
```

Comment:
It looks like this check will always be `true`. `reduce` will always return a new object so this comparison needs to be updated. Should it use `isShallowEquals` instead? It would be nice to add unit test which ensures it works properly even when I'm wrong :)

Reply:
Definitely a shallow equal :)